sound/soc/codecs/rt700-sdw.c

```diff
@@ -418,7 +418,7 @@ static int rt700_interrupt_callback(struct sdw_slave *slave,
 	dev_dbg(&slave->dev,
 		"%s control_port_stat=%x", __func__, status->control_port);
 
-	if (status->control_port & 0x4) {
+	if (status->control_port & 0x4 && !rt700->disable_workqueues) {
 		mod_delayed_work(system_power_efficient_wq,
```

Comment:
yes, I think this should work, but as @ujfalusi mentioned you need to lock the whole fragment on lines 421-424 and lock the same spin-lock when setting `rt700->disable_workqueues`

Reply:
I also think a lock around `rt700->disable_workqueues` would make this robust.

Reply:
Thanks for the feedback @ujfalusi and @lyakh 

humm, yes we can add this spinlock, but then I think we can also remove the checks in rt700.c

There would be no way for the new workqueues to be scheduled after the spinlock exit, and the cancel_work_sync would wait until the worker functions complete if they started already.